util/a2e/atoe_utils.c

```diff
@@ -555,23 +555,27 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					 * Add floating point support for dbl2str & flt2str
 					 */
 					char *tempPattern = a2e((char *)pattern, fmt - pattern);
-					size_t freeCap = (this.end > this.buffer) ? (this.end - this.buffer) : 0;
-
-					/* Extract a double from args, this works for both doubles
-					 * and floats,
-					 * NB if we use float for a single precision floating
-					 * point number the result is wrong.
-					 */
-					double argument = va_arg(args, double);
-					int requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
-					free(tempPattern);
-					CheckRet(requiredSpace);
-					if ((freeCap > 0) && (requiredSpace > 0)) {
-						char *tempAsciiValue = e2a_string(this.buffer);
-						strcpy(this.buffer, tempAsciiValue);
-						free(tempAsciiValue);
+					if (NULL != tempPattern) {
+						size_t freeCap = (this.end > this.buffer) ? (this.end - this.buffer) : 0;
+	
+						/* Extract a double from args, this works for both doubles
+						 * and floats,
+						 * NB if we use float for a single precision floating
+						 * point number the result is wrong.
+						 */
+						double argument = va_arg(args, double);
+						int requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
+						free(tempPattern);
+						CheckRet(requiredSpace);
+						if ((freeCap > 0) && (requiredSpace > 0)) {
```

Comment:
I don't think this condition is right: I think we need `freeCap > requiredSpace`:
```c
					if ((requiredSpace > 0) && (freeCap > (size_t)requiredSpace)) {
```
It's also not immediately obvious that this works properly if `freeCap == 0`: The caller may be asking how much space is required (by providing `count == 0`), but in that case `snprintf()` will return `-1` and this fails via `CheckRet()`.